lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -103,7 +103,7 @@ sub bugrefs ($self) { find_bugrefs($self->text) }
 Returns label value if C<$self> is label, e.g. 'label:my_label' returns 'my_label'
 =cut
 sub label ($self) {
-    return first { $_ !~ /^force_result/ } @{find_labels($self->text)};
+    return @{find_labels($self->text)}[0];
```

Comment:
```perl
return find_labels($self->text)->[0];
```

Reply:
I assume both would compile to the same byte-code, right? (Unfortunately there's no Perl on compiler explorer.)

Reply:
Not quite actually:
```
$ perl -MO=Terse -e '@{find_labels($self->text)}[0];'
LISTOP (0x15080eda0) leave [1]
    OP (0x15080ed70) enter
    COP (0x15080ede0) nextstate
    LISTOP (0x15080ee40) aslice
        OP (0x15080ee80) pushmark
        SVOP (0x15080eeb0) const  IV (0x150023550) 0
        UNOP (0x15080eee8) rv2av [3]
            LISTOP (0x15080ef50) leave
                OP (0x15080ef20) enter
                COP (0x15080ef90) nextstate
                UNOP (0x14fe050c0) entersub [2]
                    UNOP (0x15080eff0) null [158]
                        OP (0x14fe050f8) pushmark
                        UNOP (0x14fe05128) entersub [1]
                            OP (0x14fe051a8) pushmark
                            UNOP (0x14fe051e0) null [14]
                                SVOP (0x14fe05250) gvsv  GV (0x150023628) *self
                            METHOP (0x14fe05168) method_named  PV (0x1500235b0) "text"
                        UNOP (0x14fe05218) null [16]
                            SVOP (0x14fe05288) gv  GV (0x1500234d8) *find_labels
-e syntax OK

$ perl -MO=Terse -e 'find_labels($self->text)->[0]'
LISTOP (0x15a0c1aa0) leave [1]
    OP (0x15a0c1a70) enter
    COP (0x15a0c1ae0) nextstate
    UNOP (0x15a0c1b40) null [138]
        UNOP_AUX (0x15a0c1a30) multideref
            UNOP (0x15a0c1b80) null [135]
                UNOP (0x1596050c0) entersub [2]
                    UNOP (0x15a0c1bf0) null [158]
                        OP (0x1596050f8) pushmark
                        UNOP (0x159605128) entersub [1]
                            OP (0x1596051a8) pushmark
                            UNOP (0x1596051e0) null [14]
                                SVOP (0x159605250) gvsv  GV (0x159823628) *self
                            METHOP (0x159605168) method_named  PV (0x1598235b0) "text"
                        UNOP (0x159605218) null [16]
                            SVOP (0x159605288) gv  GV (0x1598234d8) *find_labels
-e syntax OK
```

Reply:
Not quite actually:
```
$ perl -MO=Terse -e '@{find_labels($self->text)}[0];'
LISTOP (0x15080eda0) leave [1]
    OP (0x15080ed70) enter
    COP (0x15080ede0) nextstate
    LISTOP (0x15080ee40) aslice
        OP (0x15080ee80) pushmark
        SVOP (0x15080eeb0) const  IV (0x150023550) 0
        UNOP (0x15080eee8) rv2av [3]
            LISTOP (0x15080ef50) leave
                OP (0x15080ef20) enter
                COP (0x15080ef90) nextstate
                UNOP (0x14fe050c0) entersub [2]
                    UNOP (0x15080eff0) null [158]
                        OP (0x14fe050f8) pushmark
                        UNOP (0x14fe05128) entersub [1]
                            OP (0x14fe051a8) pushmark
                            UNOP (0x14fe051e0) null [14]
                                SVOP (0x14fe05250) gvsv  GV (0x150023628) *self
                            METHOP (0x14fe05168) method_named  PV (0x1500235b0) "text"
                        UNOP (0x14fe05218) null [16]
                            SVOP (0x14fe05288) gv  GV (0x1500234d8) *find_labels
-e syntax OK

$ perl -MO=Terse -e 'find_labels($self->text)->[0]'
LISTOP (0x15a0c1aa0) leave [1]
    OP (0x15a0c1a70) enter
    COP (0x15a0c1ae0) nextstate
    UNOP (0x15a0c1b40) null [138]
        UNOP_AUX (0x15a0c1a30) multideref
            UNOP (0x15a0c1b80) null [135]
                UNOP (0x1596050c0) entersub [2]
                    UNOP (0x15a0c1bf0) null [158]
                        OP (0x1596050f8) pushmark
                        UNOP (0x159605128) entersub [1]
                            OP (0x1596051a8) pushmark
                            UNOP (0x1596051e0) null [14]
                                SVOP (0x159605250) gvsv  GV (0x159823628) *self
                            METHOP (0x159605168) method_named  PV (0x1598235b0) "text"
                        UNOP (0x159605218) null [16]
                            SVOP (0x159605288) gv  GV (0x1598234d8) *find_labels
-e syntax OK
```

Reply:
This is the kind of command I was getting at, thanks :-)

If I make sense of it correctly, using `->` should be more performant.

Reply:
Yes, it probably is.